lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -3525,4 +3525,17 @@ public function getStringLiteralQuoteCharacter()
     {
         return "'";
     }
+
+    /**
+     * Determines if a string literal is valid as an identifier in this
+     * platform.
+     *
+     * @param string $name The string literal to test
+     *
+     * @return bool
+     */
+    public function isValidIdentifier($name)
```

Comment:
This is a good point -- My intent here is only to provide a way to conditionally quote identifiers being extracted from the database. Maybe instead of a wide-ranging isValidIdentifier method here, we could name the method differently and note its intended use.

The alternative is to quote every identifier coming from the DB. This is more or less what SQL server's internal script generation does. The trick there would be to ensure things don't get double-escaped.

Right now, identifiers with illegal characters are not being properly handled when they are being retrieved from the database on SQL Server. This causes the ORM functional test suite to error out during cleanup.